numba/core/interpreter.py

```diff
@@ -2859,23 +2859,36 @@ def op_JUMP_IF_TRUE(self, inst, pred):
         self._op_JUMP_IF(inst, pred=pred, iftrue=True)
 
     def _jump_if_none(self, inst, pred, iftrue):
-        brs = {
-            True: inst.get_jump_target(),
-            False: inst.next,
-        }
-        truebr = brs[iftrue]
-        falsebr = brs[not iftrue]
+        # branch pruning assumes true falls through and false is jump
+        truebr = inst.next
+        falsebr = inst.get_jump_target()
+
+        # this seems strange
+        if not iftrue:
+            op = BINOPS_TO_OPERATORS["is"]
+        else:
+            op = BINOPS_TO_OPERATORS["is not"]
 
-        op = BINOPS_TO_OPERATORS["is"]
+        rhs = self.store(value=ir.Const(None, loc=self.loc),
+                         name=f"$constNone{inst.offset}")
+        lhs = self.get(pred)
+        isnone = ir.Expr.binop(op, lhs=lhs, rhs=rhs, loc=self.loc)
 
-        constnone = self.store(value=ir.Const(None, loc=self.loc),
-                         name="${inst.offset}constnone")
-        pred = self.get(pred)
-        isnone = ir.Expr.binop(op, lhs=pred, rhs=constnone, loc=self.loc)
+        maybeNone = f"$maybeNone{inst.offset}"
+        self.store(value=isnone, name=maybeNone)
 
-        pname = "$%spred" % (inst.offset)
-        predicate = self.store(value=isnone, name=pname)
-        branch = ir.Branch(cond=predicate, truebr=truebr, falsebr=falsebr,
+        name = f"$bool{inst.offset}"
+        gv_fn = ir.Global("bool", bool, loc=self.loc)
+        self.store(value=gv_fn, name=name)
+
+        callres = ir.Expr.call(self.get(name), (self.get(maybeNone),), (),
+                               loc=self.loc)
+
+        pname = f"$pred{inst.offset}"
```

Comment:
Note to self. Make these register names match those that are generated for < 3.11.